src/compiler/output-targets/dist-lazy/lazy-output.ts

```diff
@@ -76,6 +76,8 @@ export const outputLazy = async (config: d.Config, compilerCtx: d.CompilerCtx, b
           buildCtx.es5ComponentBundle = result.buildCtx.es5ComponentBundle;
         } else if (result.name === 'esm-browser') {
           buildCtx.esmBrowserComponentBundle = result.buildCtx.esmBrowserComponentBundle;
+          buildCtx.buildResults = result.buildCtx.buildResults;
+          buildCtx.components = result.buildCtx.components;
```

Comment:
I'm not sure why we're doing this/what's going on here - can you explain this to me please?

Reply:
Yeah! The components and buildResults are passed on to the buildCtx within this file from the resultant builds. These needed to be set on the buildCtx here because the promises don't update the buildCtx any more, and the original stats code needs to understand the state of the updated buildCtx after the promises all run. I wrote it this way so that we could keep the parallelization of the builds.

Reply:
Does this affect the memory usage of Stencil in a negative way? I mean, I'm sure there's some cost associated with it, but do you have an idea of the order of magnitude?

Reply:
I suspect this comment still holds true, and may likely improve since I brought the parallelization back. 

> I was worried that adding these to the build context would increase the memory usage significantly. Luckily that is not the case. I ran 10 builds of ionic-framework using `master` and this PR. The increase is relatively minor:
> 
> Build #	`master`	#3030
> 1	MEM: 682.1MB	MEM: 658.6MB
> 2	MEM: 743.8MB	MEM: 677.3MB
> 3	MEM: 546.5MB	MEM: 757.6MB
> 4	MEM: 727.2MB	MEM: 697.9MB
> 5	MEM: 731.8MB	MEM: 658.1MB
> 6	MEM: 730.2MB	MEM: 656.2MB
> 7	MEM: 738.5MB	MEM: 749.1MB
> 8	MEM: 726.2MB	MEM: 655.7MB
> 9	MEM: 709.2MB	MEM: 684.9MB
> 10	MEM: 663.2MB	MEM: 617.7MB
> **Average**	**MEM: 681.3MB**	**MEM: 699.9MB**